jax/_src/numpy/lax_numpy.py

```diff
@@ -3417,6 +3417,11 @@ def _take(a, indices, axis: Optional[int] = None, out=None, mode=None):
 
 def _normalize_index(index, axis_size):
   """Normalizes an index value in the range [-N, N) to the range [0, N)."""
+  try:
+    idx, ndim = operator.index(index), operator.index(axis_size)
+    return _canonicalize_axis(idx, ndim)
+  except (TypeError, ValueError):
```

Comment:
Not catching `ValueError`s here would bring the behaviour closer to numpy but would [change the current behaviour for out of bound indexing](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#out-of-bounds-indexing) resulting in inconsistent behaviour.

Reply:
In that case maybe it's cleaner to not call canonicalize axis, and rather do this:
```python
return idx + ndim if idx < 0 else idx
```
then we don't have to worry about the `ValueError`

Reply:
Also, I think a more idiomatic way to write this may be
```python
try:
  idx, ndim = operator.index(index), operator.index(axis_size)
except TypeError:
  pass
else:
  return idx + ndim if idx < 0 else idx
```
It makes clear where the TypeError is expected, and avoids catching unexpected errors.

Reply:
That's a lot cleaner. Done 👍